src/flattenObject.ts

```diff
@@ -0,0 +1,26 @@
+// Convert objecto to one level of deepness
+const flattenObject = (ob, separator = '.') => {
+  const toReturn = {}
+
+  for (const index in ob) {
+    if (!ob.hasOwnProperty || !Object.prototype.hasOwnProperty.call(ob, index)) {
+      continue
+    }
+
+    if ((typeof ob[index]) === 'object') {
```

Comment:
I don't understand this check, why do you do it?

Reply:
because it's a recursive method which only tries to flat nested structures. That's why it should only be called recursively when a current checking branch is an object...

Reply:
Makes sense

Reply:
this is the classic method that github copilot writes for you 😆